### PR TITLE
bugfixes

### PR DESCRIPTION
**Pull request by aliothe**
* check if queue cache is empty
* log stack trace if validation fails
* guard against usage of Arrays.asList

### CODE REVIEW

File: casual/casual-caller/src/main/java/se/laz/casual/connection/caller/QueueCache.java [hunk 1]
```diff
@@ -55,6 +55,12 @@ else if (cacheMap.containsKey(queueName))
                         .filter(ConnectionFactoryEntry::isValid)
                         .collect(Collectors.toList());
 
+                if(cachedForQueue.isEmpty())
+                {
+                    LOG.info(() -> "No valid connection for queuename: " + queueName);
+                    return Optional.empty();
```
Comment (TobiasLeo) [inline casual/casual-caller/src/main/java/se/laz/casual/connection/caller/QueueCache.java:61]: Should there be a test to verify proper behavior when no valid entry exists for a given queue in the cache?

Comment (aliothe) [reply casual/casual-caller/src/main/java/se/laz/casual/connection/caller/QueueCache.java:61]: Looks like there are 0 tests for QueueCache to begin with, was wondering why I had not added a test for that

Comment (aliothe) [reply casual/casual-caller/src/main/java/se/laz/casual/connection/caller/QueueCache.java:61]: Seems it is tested via the CacheTest, I'll add a test for this as well there

Comment (aliothe) [reply casual/casual-caller/src/main/java/se/laz/casual/connection/caller/QueueCache.java:61]: Done

**Review by TobiasLeo — APPROVED** [review]

**Pull request merged**